lib/shard/src/segment_holder/mod.rs

```diff
@@ -70,6 +71,11 @@ pub struct SegmentHolder {
 
     /// Holds the first uncorrected error happened with optimizer
     pub optimizer_errors: Option<String>,
+
+    /// A special segment version that is usually used to keep track of manually bumped segment versions.
+    /// An example for this are operations that don't modify any points but could be expensive to recover from during WAL recovery.
+    /// To acknowledge them in WAL, we overwrite the max_persisted value in `Self::flush_all` with the segment version stored here.
+    max_persisted_segment_version_overwrite: AtomicOptionU64,
```

Comment:
Ah, I didn't consider that we need mutable access to the holder (if not using atomics or others).

Using an atomic is fine, but I think that `AtomicOptionU64` may be a bit over engineered.

What about a regular AtomicU64 that is not optional, being 0 by default? Having 0 by default will not be an issue in the flush stage when taking the max version across segments.

Wdyt?